pages_/my-shop.tsx

```diff
@@ -265,15 +269,15 @@ const MyShop = ({ isLoggedIn, shopId, encodedShopId }: Props) => {
 export const getServerSideProps: GetServerSideProps = async (context) => {
   const token = getToken(context);
 
-  if (!token || !token.shopId) {
-    return { props: { isLoggedIn: false } };
+  if (!token || !token?.isValid || !token.clientId) {
+    return { props: { isLoggedIn: false, isShopCreated: false, shopId: null } };
   }
 
   return {
     props: {
       isLoggedIn: true,
-      shopId: token.shopId,
-      encodedShopId: encodeId(token.shopId),
+      shopId: token.shopId === undefined ? null : token.shopId,
+      encodedShopId: encodeId(token.shopId || 0),
```

Comment:
we should just return one encoded shopId prop, it doesn't make sense to return the two ids. If you just call `encodeId(token.shopId)` and `shopId` is null, it would return and empty string, and that's enough to pass as a prop.

Reply:
sounds good